### PR TITLE
Testing EndpointResults without Dispatcher

### PR DESCRIPTION
**Pull request by vkostyukov**
I only did `BodySpec` to get an idea of how does it look and feel. I will carry on if folks are ok with this approach.

### CODE REVIEW

File: core/src/main/scala/io/finch/EndpointResult.scala [hunk 1]
```diff
@@ -86,9 +59,19 @@ object EndpointResult {
...
-    final def output: Option[F[Output[A]]] = self match {
-      case EndpointResult.Matched(_, _, out) => Some(out)
-      case _                                 => None
+    final def outputAttempt(implicit F: ApplicativeError[F, Throwable]): F[Either[Throwable, Output[A]]] =
+      F.attempt(output)
+
+    final def output: F[Output[A]] = self match {
+      case EndpointResult.Matched(_, _, out) => out
+      case _                                 =>
+        throw new IllegalStateException("Endpoint didn't match")
```
Comment (tpetillot) [inline core/src/main/scala/io/finch/EndpointResult.scala:68]: I would rather raise an error than throw it.
```suggestion
      case _                                 => F.raiseError(new IllegalStateException("Endpoint didn't match"))
```

**Review by tpetillot — COMMENTED** [review]
I really like what you are proposing!

**Comment by joroKr21** [comment]
I like the approach too 👍 - much simpler!

**Comment by joroKr21** [comment]
What about `EntityEndpointLaws`? It looks like it's not so easy.

**Comment by tpetillot** [comment]
If you still want to define a rule set, either we still pass the dispatcher on this one, or, we defined `EntityEndpointLaws` on `IO` as all spec is defined on IO?

**Pull request closed without merging**